presentation/src/main/java/com/knocklock/presentation/lockscreen/LockScreenStateHolder.kt

```diff
@@ -1,83 +1,109 @@
-import androidx.compose.runtime.rememberCoroutineScope
-import com.knocklock.domain.usecase.notification.DeleteAllNotificationUseCase
-import com.knocklock.domain.usecase.notification.GetNotificationUseCase
-import dagger.hilt.EntryPoint
-import dagger.hilt.InstallIn
-import dagger.hilt.android.EntryPointAccessors
-import dagger.hilt.components.SingletonComponent
-import kotlinx.coroutines.CoroutineScope
-import kotlinx.coroutines.coroutineScope
 import kotlinx.coroutines.flow.MutableStateFlow
 import kotlinx.coroutines.flow.asStateFlow
-import kotlinx.coroutines.launch
+import java.util.*
 import javax.inject.Inject
 
 /**
  * @Created by 김현국 2023/01/03
  * @Time 3:26 PM
  */
 
-@EntryPoint
-@InstallIn(SingletonComponent::class)
-interface UseCaseEntryPoint {
-    fun getNotificationUseCase(): GetNotificationUseCase
-    fun deleteAllNotificationUseCase(): DeleteAllNotificationUseCase
-}
-
+@Stable
 class LockScreenStateHolder @Inject constructor(
-    context: Context,
-    private val coroutineScope: CoroutineScope
+    context: Context
 ) {
 
-    private val _notificationList: MutableStateFlow<NotificationUiState> = MutableStateFlow(NotificationUiState.Empty)
+    private val _notificationList: MutableStateFlow<NotificationUiState> = MutableStateFlow(
+        NotificationUiState.Empty
+    )
     val notificationList = _notificationList.asStateFlow()
 
-    private val useCaseEntryPoint =
-        EntryPointAccessors.fromApplication(
-            context,
-            UseCaseEntryPoint::class.java
-        )
+    private val packageManager by lazy { context.packageManager }
 
-    val getNotificationUseCase = useCaseEntryPoint.getNotificationUseCase()
-    val deleteAllNotificationUseCase = useCaseEntryPoint.deleteAllNotificationUseCase()
+    fun updateNotificationArray(notificationArray: Array<StatusBarNotification>) {
+        val notificationUiState = NotificationUiState.Success(
+            notificationList = notificationArray.map { statusBarNotification ->
+                with(statusBarNotification.notification.extras) {
+                    var appTitle = ""
+                    val subText: String = convertString(getCharSequence("android.subText"))
+                    val title: String = convertString(getCharSequence("android.title"))
+                    val content: String = convertString(getCharSequence("android.text"))
+                    val packageName = statusBarNotification.packageName
+                    val date = Date(statusBarNotification.postTime)
+                    val hour = DateFormat.format("HH", date).toString().toInt()
+                    val stringPostTime = DateFormat.format("HH:mm", date).toString()
+                    val stringPostTimeBuilder = StringBuilder()
+                    if (hour <= 12) {
+                        stringPostTimeBuilder.append("오전 $stringPostTime")
+                    } else {
+                        stringPostTimeBuilder.append("오후 $stringPostTime")
+                    }
```

Comment:
```suggestion
                    val date = Date(statusBarNotification.postTime)
                     val stringPostTime = try {
                              SimpleDateFormat("a HH:mm", Locale.KOREA).format(date)
                          } catch (e: Exception) {
                              ""
                          }
```

오전 오후도 시간 포맷으로 나타낼 수 있을 것 같아요!

Reply:
@jaeryo2357 이런 꿀기능이 있었네영 [b2ddc67](https://github.com/Knock-Lock/Knock-Lock/pull/75/commits/b2ddc67d201a44f3000540d56d3370b2dd1d100e)추가했습니다